siliconcompiler/report/Home.py

```diff
@@ -0,0 +1,601 @@
+    """
+    nodes = []
+    edges = []
+    opacity = 0.2
+    width = 1
+    for step, index in node_dependencies:
+        if (step, index) in successful_path:
+            opacity = 1
+            if (step, index) in chip._get_flowgraph_exit_nodes() or \
+               (step, index) in chip._get_flowgraph_entry_nodes():
+                width = 3
+
+        flow = chip.get("option", "flow")
+        task_status = chip.get('flowgraph', flow, step, index, 'status')
+        if task_status == TaskStatus.SUCCESS:
+            node_color = success_color
+        elif task_status == TaskStatus.ERROR:
+            node_color = failure_color
+        else:
+            node_color = pending_color
+
+        tool, task = chip._get_tool_task(step, index)
+        label = step + index + "\n" + tool + "/" + task
+        if chip._is_builtin(tool, task):
+            label = step + index + "\n" + tool
+
+        nodes.append(Node(id=step + index,
+                          label=label,
+                          color=node_color,
+                          opacity=opacity,
+                          borderWidth=width,
+                          shape='oval'))
+        width = 3
+        for source_step, source_index in node_dependencies[step, index]:
+            if (source_step, source_index) in successful_path and \
+               (source_step, source_index) in successful_path:
+                width = 5
+            edges.append(Edge(source=source_step + source_index,
+                              dir='up',
+                              target=step + index,
+                              width=width,
+                              color='black',
+                              curve=True))
+
+    return nodes, edges
+
+
+def show_file_preview(display_file_content):
+    """
+    Displays the file_preview if present. If not, displays an error message.
+
+    Args:
+        display_file_content (bool) : boolean representing if we should (true)
+                                      or should not (false) display content
+    """
+    if display_file_content:
+        path = streamlit.session_state['selected'][0]
+        file_name, file_extension = os.path.splitext(path)
+        header_col, download_col = streamlit.columns([0.89, 0.11], gap='small')
+        with header_col:
+            streamlit.header('File Preview')
+        with download_col:
+            streamlit.markdown(' ')  # aligns download button with title
+            streamlit.download_button(label="Download file",
+                                      data=path,
+                                      file_name=path[path.rfind("/"):])
+
+        if file_extension == ".png":
+            streamlit.image(path)
+        else:
+            with open(path, 'r') as file:
+                content = file.read()
+            if file_extension == ".json":
+                streamlit.json(content)
+            else:
+                streamlit.code(content, language='markdwon', line_numbers=True)
```

Comment:
markdwon? what are the options? should we expand this fully? by extensions?

Reply:
Yes, we definitely could expand this to all other options.